src/main/java/dev/tycho/stonks/scheduledtasks/SubscriptionCheckTask.java

```diff
@@ -0,0 +1,112 @@
+package dev.tycho.stonks.scheduledtasks;
+
+import dev.tycho.stonks.Stonks;
+import dev.tycho.stonks.managers.DatabaseHelper;
+import dev.tycho.stonks.model.logging.Transaction;
+import dev.tycho.stonks.model.service.Service;
+import dev.tycho.stonks.model.service.Subscription;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.Player;
+
+import java.sql.SQLException;
+import java.util.List;
+
+public class SubscriptionCheckTask implements Runnable {
+
+  public SubscriptionCheckTask() {
+    System.out.println("New subscription payment task scheduled");
+  }
+
+  @Override
+  public void run() {
+    System.out.println("Automatically renewing subscriptions");
+    //We want to go through all subscriptions
+    List<Subscription> subscriptions;
+    try {
+      subscriptions = DatabaseHelper.getInstance().getDatabaseManager().getSubscriptionDao().queryForAll();
+    } catch (SQLException e) {
+      e.printStackTrace();
+      Bukkit.broadcastMessage("Error updating subscriptions");
+      return;
+    }
+
+    for (Subscription subscription : subscriptions) {
+      renewSubscription(subscription);
+      cancelSubscriptionIfOverdue(subscription);
+    }
+  }
+
+  private void renewSubscription(Subscription subscription) {
+    Service service = subscription.getService();
+    OfflinePlayer player = Bukkit.getOfflinePlayer(subscription.getPlayerId());
+    Player onlinePlayer = null;
+    if (player.isOnline()) onlinePlayer = player.getPlayer();
+
+    if (subscription.isOverdue() && subscription.isAutoPay()) {
+      System.out.println("Auto billing " + player.getName() + " $" + service.getCost() + " for service id " + service.getId());
+      //Subscription is overdue and we must renew it
+      if (onlinePlayer != null)
+        onlinePlayer.sendMessage(ChatColor.YELLOW + "Your subscription for " + service.getName() + " has ended." + ChatColor.GREEN + " Automatically renewing it now.....");
+
+      //Try and charge them the amount due
+      if (!Stonks.economy.withdrawPlayer(player, service.getCost()).transactionSuccess()) {
+        //Player does not have enough money
+        System.out.println("(failed - they didn't have enough money)");
+        if (onlinePlayer != null)
+          onlinePlayer.sendMessage(ChatColor.RED + "Failed to renew. You don't have enough money, please get more then manually resubscribe");
```

Comment:
You should copy the `sendMessage(Player, String)` method to here and use it for this